### PR TITLE
feat(cell-based-table): compute value meta on distribution keys instead of pk

### PR DESCRIPTION
**Pull request by xx01cyx**
## What's changed and what's your intention?

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

### Summarize your change (**mandatory**)

- Add distribution keys in frontend to proto in MV node and arrange node. (By congyi)
- Compute value meta on distribution keys instead of pk in cell based table.
- Set mv distribution key indices correctly.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

File: src/common/src/array/data_chunk_iter.rs [hunk 1]
```diff
@@ -288,6 +288,26 @@ impl Row {
...
+        H: BuildHasher,
+    {
+        let mut hasher = hash_builder.build_hasher();
+        for idx in hash_indices {
+            let datum = self.0.get(*idx);
+            match datum {
+                Some(datum) => datum.hash(&mut hasher),
+                None => {
+                    return Err(
+                        ErrorCode::InternalError(format!("index {} out of row bound", idx)).into(),
```
Comment (BowenXiao1999) [inline src/common/src/array/data_chunk_iter.rs:304]: Is this Error expected to happen? Dist key should always included by pk right.

**Review by BowenXiao1999 — COMMENTED** [review]
Rest LGTM

File: src/common/src/array/data_chunk_iter.rs (hunk 1 above)
Comment (xx01cyx) [reply src/common/src/array/data_chunk_iter.rs:304]: The error is unwrapped in the caller of this function. The hash function is more like a util function and shouldn't panic.

**Review by fuyufjh — APPROVED** [review]
LGTM

**Pull request merged**